src/settings.rs

```diff
@@ -29,20 +29,46 @@ use postcard::ser_flavors::Flavor;
 use stm32h7xx_hal::flash::LockedFlashBank;
 
 #[derive(Clone, miniconf::Tree)]
-pub struct Settings {
+pub struct Settings<C, const Y: usize>
+where
+    for<'d> C: miniconf::JsonCoreSlash<'d, Y>
+        + Default
+        + Clone
+        + serde::Serialize
+        + serde::Deserialize<'d>
+        + 'static,
+{
     pub broker: heapless::String<255>,
     pub id: heapless::String<23>,
+
+    pub runtime: C,
```

Comment:
@jordens This is fundamentally the problem. The `Serialize` and `Deserialize` bounds shouldn't be required here because `runtime` should be recursive tree of depth `Y`. We might be able to work around this by pulling this entire structure out into the final application-level `bin` and just require something that implements serial_settings::Settings instead.

Reply:
I don't get it yet.

For what you do in `reload()` `C` does need to be `Serialize+Deserialize`. That's fine.

Reply:
The idea is that the runtime settings should be individually accessible as leaf nodes for each setting, not that they should be saved as a single blob. I.e. currently, you'd have to set the entire runtime settings structure with a single USB request:
```
> set runtime="{"afe": ["G1", "G10"], "iir_ch: [...], "allow_hold": False, ...}"
```

The actual goal would be to allow the settings to individually be configurable as:
```
> set runtime/afe/0="G1"
```

That way, if we modify the runtime settings in code, all of the existing settings in flash can be retained and the new setting can be defaulted.